candle-transformers/src/models/repvgg.rs

```diff
@@ -0,0 +1,233 @@
+//! RepVGG inference implementation
+//!
+//! See "RepVGG: Making VGG-style ConvNets Great Again" Ding et al. 2021
+//! https://arxiv.org/abs/2101.03697
+
+use std::collections::HashMap;
+
+use candle::{Result, Tensor, D};
+use candle_nn::{
+    batch_norm, conv2d_no_bias, linear, BatchNorm, Conv2d, Conv2dConfig, Func, VarBuilder,
+};
+
+#[derive(Clone)]
+struct RepVGGConfig {
+    a: f32,
+    b: f32,
+    groups: usize,
+    stages: [usize; 4],
+}
+
+#[rustfmt::skip]
+fn repvgg_config(cfg: &str) -> RepVGGConfig {
+    let configs = HashMap::from([
+        ( "A0", RepVGGConfig { a: 0.75, b: 2.5, groups: 1, stages: [2, 4, 14, 1]}),
```

Comment:
Let's use an enum instead of a string for the model variants.

Reply:
I used strings here because of the way I used strings in main.rs too, where the command line --which arg is being checked by clap and then only valid values are being passed as model config. I did this to avoid the  matches for each of the variants and be able to derive the model filename via a single line of string formatting. Shall I change main.rs too?

Reply:
Re the model filename, you can certainly have a `as_str()` method on the enum that would return `"A0"` etc (or `model_filename()`, ...
I think ideally the variant would be defined on the example side so that the clap macro for which can be applied to it - you'll find lots of examples of this in the examples directory, and the `repvgg.rs` file would define `RepVGGConfig::a0()`, etc. Similar a bit to what is done [here for vgg](https://github.com/huggingface/candle/blob/d3bdd788cfdcf49b6ea539b77647b82a0b979db0/candle-examples/examples/vgg/main.rs#L12).

Reply:
Made the final tweaks for this so that this PR can be merged, thanks!